src/antlr/Boa.g

```diff
@@ -202,7 +203,12 @@ outputType returns [OutputType ast]
 	@after { $ast.setPositions($l, $c, getEndLine(), getEndColumn()); }
 	: OUTPUT (tk=SET { $ast = new OutputType((Identifier)new Identifier($tk.text).setPositions(getStartLine($tk), getStartColumn($tk), getEndLine($tk), getEndColumn($tk))); } | id=identifier { $ast = new OutputType($id.ast); }) (LPAREN el=expressionList RPAREN { $ast.setArgs($el.list); })? (LBRACKET m=component RBRACKET { $ast.addIndice($m.ast); })* OF m=component { $ast.setType($m.ast); } (WEIGHT m=component { $ast.setWeight($m.ast); })? (FORMAT LPAREN el=expressionList RPAREN)?
 	;
-
+modelType returns [ModelType ast]
+	locals [int l, int c]
+	@init { $l = getStartLine(); $c = getStartColumn(); }
+	@after { $ast.setPositions($l, $c, getEndLine(), getEndColumn()); }
+	: id=identifier { $ast = new ModelType($id.ast); } OF m=component { $ast.setType($m.ast); }
```

Comment:
identifier? so users can write *anything* here and it will accept it... that seems too weak of a parser rule

Reply:
I think that it only accepts model types because I have to load the trained model.

model: LinearRegression of {int,int};
model = load("/home/tess/Desktop/datagen/0/train1587614590388ML.model");

Reply:
I have fixed the grammar, it will be model: Model of LinearRegression of {int,int};.
I also add an exception in Typechecker to raise the error if someone puts wrong statement as you mentioned.